app/views/layouts/application.html.erb

```diff
@@ -105,8 +105,30 @@
   </div>
 </div>
 <% unless internal_navigation? %>
+  <script>
```

Comment:
I tried shifting this code to `initializeFooter.js` file in `app/javascript/packs/initializers` and then using it in `app/javascript/packs/baseInitializers.js` but this was not working perfectly, on going back to home page it was showing footer again.

Let me know if you have any suggestions.

Reply:
I think part of why it's not working is that InstantClick handles most of our internal navigation so you would need to call the function in the InstantClick 'change' listener in `baseInitializers` too?

Reply:
Hi @filleduchaos 
I actually tried that earlier and it was not working. I tried something like this with the latest code:

Code for `app/javascript/packs/initializers/initializeFooter.js`
```
export function initializeFooter() {
	function toggleFooterVisibility() {
		const footer = document.querySelector("#footer");
		const pathsToHideFooter = ['/', '/top/week', '/top/month', '/top/year', '/top/infinity', '/latest'];
		if (!footer) {
			return;
		}
		if (pathsToHideFooter.some(path => window.location.pathname === path)) {
			footer.classList.add('hidden');
		} else {
			footer.classList.remove('hidden');
		}
	}

	// To ensure correct footer visibility even when opening a child page in a new tab,
	// we add a DOMContentLoaded event listener, which calls toggleFooterVisibility
	// after the entire page, including the footer, is fully loaded.
	// document.addEventListener("DOMContentLoaded", () => {
	// 	toggleFooterVisibility();
	// });
	toggleFooterVisibility();
}

```

And in `baseInitializers` called this function globally as well as in `InstantClick` `change`.


This fails on one very specific case:
1. From home page, visit any post page.
2. Now from post page, click back. to visit home page.
3. Scroll to the bottom to see the footer.

Reply:
Ah, it's probably best to just leave it here then. I tried it out but I couldn't figure out a fix for it either

Reply:
Thanks a lot @filleduchaos